my-app/src/app/app.component.spec.ts

```diff
@@ -9,7 +10,8 @@ describe('AppComponent', () => {
         RouterTestingModule
       ],
       declarations: [
-        AppComponent
+        AppComponent,
+        MainLayoutComponent
```

Comment:
you may use NO_ERROR_SCHEMA or CUSTOM_ELEMENTS_SCHEMA or fake implementation of components, directives and pipes
instead of declaring all components.
As far as unit testing means while testing the component, you should not depend on other components or services or smth like that

Reply:
understood. thanks for the tip